src/cloudevent/mocks/database/helpers.ts

```diff
@@ -0,0 +1,101 @@
+import { CloudFunction, database } from 'firebase-functions/v2';
+import { DeepPartial } from '../../types';
+import {
+  exampleDataSnapshot,
+  exampleDataSnapshotChange,
+} from '../../../providers/database';
+import { getBaseCloudEvent } from '../helpers';
+import { Change } from 'firebase-functions';
+
+export function getDatabaseSnapshotCloudEvent(
+  cloudFunction: CloudFunction<database.DatabaseEvent<any>>,
+  cloudEventPartial?: DeepPartial<database.DatabaseEvent<any>>
+) {
+  const {
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  } = getCommonDatabaseFields(cloudFunction, cloudEventPartial);
+
+  const data =
+    (cloudEventPartial?.data as database.DataSnapshot) ||
+    exampleDataSnapshot(ref);
+
+  return {
+    // Spread common fields
+    ...getBaseCloudEvent(cloudFunction),
+
+    // Update fields specific to this CloudEvent
+    data,
+
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  };
+}
+
+export function getDatabaseChangeSnapshotCloudEvent(
+  cloudFunction: CloudFunction<
+    database.DatabaseEvent<Change<database.DataSnapshot>>
+  >,
+  cloudEventPartial?: DeepPartial<
+    database.DatabaseEvent<Change<database.DataSnapshot>>
+  >
+): database.DatabaseEvent<Change<database.DataSnapshot>> {
+  const {
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  } = getCommonDatabaseFields(cloudFunction, cloudEventPartial);
+
+  const data =
+    (cloudEventPartial?.data as Change<database.DataSnapshot>) ||
+    exampleDataSnapshotChange(ref);
+
+  return {
+    // Spread common fields
+    ...getBaseCloudEvent(cloudFunction),
+
+    // Update fields specific to this CloudEvent
+    data,
+
+    instance,
+    firebaseDatabaseHost,
+    ref,
+    location,
+    params,
+  };
+}
+
+export function getCommonDatabaseFields(
+  cloudFunction: CloudFunction<database.DatabaseEvent<any>>,
+  cloudEventPartial?: DeepPartial<database.DatabaseEvent<any>>
+) {
+  const instance =
+    (cloudEventPartial?.instance as string) ||
+    cloudFunction.__endpoint.eventTrigger.eventFilters.instance ||
+    'instance-1';
+  const firebaseDatabaseHost =
+    (cloudEventPartial?.firebaseDatabaseHost as string) ||
+    'firebaseDatabaseHost';
+  const ref =
+    (cloudEventPartial?.ref as string) ||
+    cloudFunction?.__endpoint?.eventTrigger?.eventFilterPathPatterns?.ref ||
```

Comment:
Is it okay if this `ref` looks like `foo/{something}/bar`? We use the actual CloudEvent to figure out the capture variable. I think it's fine since a user wouldn't be testing this, or at least I wouldn't think they would

Reply:
I think that entirely depends on what end-users would expect.
We could maybe inspect the user supplied `cloudEventPartial` to see if there are params to full in the blanks.

@inlined thoughts?

Reply:
The v1 code resolves actual params (e.g. given a trigger of "/users/{uid}" the event would have "/users/123456" and the event would have params {"uid": "123456"}. We should do the same.

Reply:
Updated to do the same, as well as updated to handle single and multi capture.